applications/typechain-polkadot-follow-up-2.md

```diff
@@ -0,0 +1,221 @@
+- **FTE:** 2.5
+- **Costs:** 52,500 USD
+
+| Number | Deliverable | Specification |
+| ------ | ----------- | ------------- |
+| 0a | License | MIT |
+| 0b | Documentation | Cover new-added features in documentation and usage examples. Prepare generated code to have more informative IDE hints based on TSDoc and the output type system (if needed). |
+| 1 | Investigation & Refactoring | Broaden types definitions for methods arguments and return values (to full coverage). Also, refactor project structure to monorepo for future development | 
+| 2 | Parser & generators modules | Design and implement a new parser module for ABI JSON to work with different versions of the ABI. Parser's output structure serves as an input for generators. Refactor, replace inline generation with the parser to generator flow. |
+| 3 | Contract deployment | Add availability to deploy contracts with `Constructors` field, using `*.contract` files.
+| 4a | Contract classes extension | Extend generated contract classes with valuable properties ordinarily available on the contract (e.g., address, name, signer, etc.). Also, provide the ability to change signer and contract address easily without creating a new contract manually. |
+| 4b | Methods' names | Format methods' names in the output from the original `MethodTrait::method_name` to a more user-friendly `methodName` naming scheme while resolving overlap in formatted names. |
+| 5 | Testing | Complete coverage of PSP22 contract with integration tests. Both for contract usage and deployment. We will be testing arguments' & return values' types' correctness. |
+| 6 | Article | We will publish an article that explains features of Typechain, how to use it in projects |
+| 7 | Branding | Make logotype for typechain and better README |
+
+## Current work - Scope of this grant
+### Milestone 3 - High-level improvements, flexibility and simplifying of usage
+
+- **Estimated duration:** 8 weeks
+- **FTE:** 2.5
+- **Costs:** 56,000 USD
+
+
+| Number | Deliverable | Specification |
+| ------ | ----------- | ------------- |
+| 0a. | License | MIT | 
+| 0b. | Documentation | We will cover new-added features in documentation and usage examples. | 
+| 0c. | Testing and Teesting Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. | 
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. | 
+| 0e. | Article | We will publish article that explain how to connect it to the project and describe the types of connection options (directly or via compiler). | 
+| 1. | User-defined plugins | We will provide an opportunity for users to make their plugins, for instance, how to use parsed types, adding new fields and so on. | 
+| 2. | Subscribing to events | We will research and subscribe to events the contract emits, which will be handy for developers. |
```

Comment:
What library (if any) are you planning to use here? For example, [RxJs Observables](https://rxjs.dev/guide/observable) might be a good fit here.